File: PO PROJEKT 1/PO PROJEKT 1/PROJEKT PO PLIKI ŹRÓDŁOWE/Nightshade.cpp
#include "Nightshade.h"

Nightshade::Nightshade(World* world) {
	increaseStrength(0);
	setInitiative(0);
	sprite = 'n';
	this->world = world;
	spawn();
}

Nightshade::Nightshade(World* world, Coordinate coord) {
	increaseStrength(0);
	setInitiative(0);
	sprite = 'n';
	this->world = world;
	world->board.insert({ coord, this });
}


Organism* Nightshade::createChild(Coordinate coord) {
	return new Nightshade(world, coord);
}

bool Nightshade::poisonous(Coordinate coord, Coordinate newCoord) {
	cout << world->board[coord]->getSprite() << " zjada trujaca rosline i umiera :(" << endl;
	world->board.erase(coord);
	world->board.erase(newCoord);
	return true;
}

void Nightshade::draw() {
	cout << GREEN_BCKG << BLACK << getSprite() << RESET;
}

File: Symulator Świata/nightshade.py
from plant import Plant


class Nightshade(Plant):

    def __init__(self, world, coord=None):
        self.name = 'n'
        self._strength = 99
        self._initiative = 0
        self._age = 0
        self.sprite = (50, 50, 120)
        self.world = world
        if coord is not None:
            self.world.board[coord] = self
        else:
            self.spawn()

    def poisonous(self, coord, newCoord):
        self.world.commentator.addComment("{0} zjada trujaca rosline".format(self.world.board[coord].name))
        self.world.board.pop(coord)
        self.world.board.pop(newCoord)
        return True

    def createChild(self, coord):
        n = Nightshade(self.world, coord)